isa.c

```diff
@@ -77,7 +77,7 @@ static void decode_and_exec(uint32_t instruction)
                 exec_STRB(instruction);
             }
         }
-    } else if (get_bits(instruction, 27, 26) ==  0x0) {
+    } else if (get_bits(instruction, 27, 26) ==  0x0 && get_bit(instruction, 4) == 0) {
```

Comment:
there are three types of data instructions, this just covers the first one. but anyway i'm merging your pr, i'll fix this after merge